File: leetcode_problems/typescript/src/lc2956_find_common_elements_between_two_arrays.ts
function findIntersectionValues(nums1: number[], nums2: number[]): number[] {
  let m1: Map<number, number> = new Map();
  let m2: Map<number, number> = new Map();
  for (let num of nums1) {
    m1.set(num, (m1.get(num) || 0) + 1);
  }
  for (let num of nums2) {
    m2.set(num, (m2.get(num) || 0) + 1);
  }

  let r1 = 0;
  let r2 = 0;
  m1.forEach((v, k) => {
    if ((m2.get(k) || 0) > 0) {
      r1 += v;
    }
  });
  m2.forEach((v, k) => {
    if ((m1.get(k) || 0) > 0) {
      r2 += v;
    }
  });

  return [r1, r2];
}

File: leetcode_problems/cpp/lc2956_find_common_elements_between_two_arrays.cpp
#include <vector>
#include <iostream>
#include <unordered_map>
#include <unordered_set>
using std::unordered_map;
using std::unordered_set;
using std::vector;

class Solution
{
public:
    vector<int> findIntersectionValues(vector<int> &nums1, vector<int> &nums2)
    {
        unordered_map<int, int> m1;
        unordered_map<int, int> m2;
        for (int &num : nums1)
        {
            m1[num]++;
        }
        for (int &num : nums2)
        {
            m2[num]++;
        }
        int r1 = 0;
        int r2 = 0;
        for (auto &pair : m1)
        {
            int k = pair.first;
            int v = pair.second;
            if (m2[k] > 0)
            {
                r1 += v;
            }
        }
        for (auto &pair : m2)
        {
            int k = pair.first;
            int v = pair.second;
            if (m1[k] > 0)
            {
                r2 += v;
            }
        }
        return {r1, r2};
    }
};

int main()
{
    return 0;
}